ports/cpp/source/antlr4-c3/CodeCompletionCore.hpp

```diff
@@ -110,44 +129,49 @@ class CodeCompletionCore {
    * Tailoring of the result:
    * Tokens which should not appear in the candidates set.
    */
-  std::unordered_set<size_t> ignoredTokens;  // NOLINT: public field
+  void setIgnoredTokens(std::unordered_set<size_t>&& ignoredTokens);
 
   /**
    * Rules which replace any candidate token they contain.
    * This allows to return descriptive rules (e.g. className, instead of
    * ID/identifier).
    */
-  std::unordered_set<size_t> preferredRules;  // NOLINT: public field
+  void setPreferredRules(std::unordered_set<size_t>&& preferredRules);
 
   /**
    * Specify if preferred rules should translated top-down (higher index rule
    * returns first) or bottom-up (lower index rule returns first).
    */
-  bool translateRulesTopDown = false;  // NOLINT: public field
-
-  // --------------------------------------------------------
-  // Debugging Options
-  // --------------------------------------------------------
-  // Print human readable ATN state and other info.
+  void setTranslateRulesTopDown(bool isEnabled);
 
-  /** Not dependent on showDebugOutput. Prints the collected rules + tokens to
-   * terminal. */
-  bool showResult = false;  // NOLINT: public field
-
-  /** Enables printing ATN state info to terminal. */
-  bool showDebugOutput = false;  // NOLINT: public field
-
-  /** Only relevant when showDebugOutput is true. Enables transition printing
-   * for a state. */
-  bool debugOutputWithTransitions = false;  // NOLINT: public field
-
-  /** Also depends on showDebugOutput. Enables call stack printing for each rule
-   * recursion. */
-  bool showRuleStack = false;  // NOLINT: public field
-
-  // --------------------------------------------------------
-  // Usage
-  // --------------------------------------------------------
+  /**
+   * Print human readable ATN state and other info.
+   * NB. This API is only for a debugging and so is volatile.
+   */
+  struct {
+    /**
+     * Not dependent on showDebugOutput.
+     * Prints the collected rules + tokens to terminal.
+     */
+    bool result = false;
+
+    /**
+     * Enables printing ATN state info to terminal.
+     */
+    bool debugOutput = false;
+
+    /**
+     * Only relevant when showDebugOutput is true.
+     * Enables transition printing for a state.
+     */
+    bool debugOutputWithTransitions = false;
+
+    /**
+     * Also depends on showDebugOutput.
+     * Enables call stack printing for each rule recursion.
+     */
+    bool ruleStack = false;
+  } show;  // NOLINT: public field, only for debugging
```

Comment:
Putting all debug options in an own struct is ok, just don't name it `show`. Additionally type names have to be in PascalCase, so a natural choice here would be: `DebugOptions` instead of `show` and it would be passed in as parameter like `(..., DebugOptions const& debugOptions)...`